spec/unit/remote_cache_spec.rb

```diff
@@ -0,0 +1,457 @@
+RSpec.describe ThreeScaleToolbox::RemoteCache do
+  let(:proxied_object) { double }
+  let(:service_id) { 1 }
+  let(:backend_id) { 7654 }
+  let(:metrics) { [{'id' =>  1}, {'id' =>  2}] }
+  let(:methods) { [{'id' =>  2}] }
+  let(:error_response) { {'errors' => {}} }
+  subject { described_class.new(proxied_object) }
+
+  it 'list_metrics first call cache miss' do
+    expect(proxied_object).to receive(:list_metrics).and_return(metrics)
+    expect(subject.list_metrics(service_id)).to eq(metrics)
+  end
+
+  it 'list_metrics second call cache hit' do
+    expect(proxied_object).to receive(:list_metrics).and_return(metrics)
```

Comment:
Shouldn't this to receive "once" ?

Reply:
it is expecting to receive one time only. The second call would fail. Unless you change to something like this:
```
expect(proxied_object).to receive(:list_metrics).twice.and_return(metrics)
```

or

```
allow(proxied_object).to receive(:list_metrics).and_return(metrics)
```